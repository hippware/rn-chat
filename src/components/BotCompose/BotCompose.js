// @flow

import React from 'react';
import {Alert, View, Image, StyleSheet, TouchableOpacity} from 'react-native';
import {observer} from 'mobx-react/native';
import {observable} from 'mobx';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
import {k, width} from '../Global';
import {colors} from '../../constants';
import locationStore from '../../store/locationStore';
import Bot from '../../model/Bot';
import botStore from '../../store/botStore';
import Screen from '../Screen';
import Button from '../Button';
import * as log from '../../utils/log';
import {RText} from '../common';
import EditControls from './EditControls';
import ComposeCard from './ComposeCard';
import {showImagePicker} from '../ImagePicker';
import {Actions} from 'react-native-router-flux';

const TRANS_WHITE = colors.addAlpha(colors.WHITE, 0.75);

type Props = {
  bot: Bot,
  isFirstScreen?: boolean,
  edit?: boolean,
  titleBlurred?: boolean,
};

@observer
class BotCompose extends React.Component<Props> {
  @observable keyboardHeight: number = 0;
  @observable isLoading: boolean = false;
  card: any;

  setLoading = (s: boolean) => (this.isLoading = s);

  save = async () => {
    const {bot} = this.props;
    if (!bot.title) {
      Alert.alert('Title cannot be empty');
      this.card && this.card.focusTitle();
      return;
    }
    try {
      this.isLoading = true;

      const {isNew} = botStore.bot;
      await botStore.save();

      if (isNew) {
        Actions.pop({animated: false});
        Actions.pop();
        setTimeout(() => Actions.botDetails({item: botStore.bot.id, isNew: true}));
      } else {
        Actions.pop();
      }
    } catch (e) {
      Alert.alert('There was a problem saving your bot');
      console.error(e);
    } finally {
      this.isLoading = false;
    }
  };

  onCoverPhoto = (): void => {
    if (!this.props.isFirstScreen) {
      showImagePicker('', (source, response) => {
        botStore.setCoverPhoto({source, ...response});
      });
    }
  };

  render() {
    const {bot, isFirstScreen, edit, titleBlurred} = this.props;
    if (!bot) {
      log.log('NO BOT IS DEFINED', {level: log.levels.ERROR});
      return <Screen isDay={locationStore.isDay} />;
    }
    const saveable = bot.title.length > 0 && bot.location && bot.address;

    return (
      <Screen isDay={locationStore.isDay}>
        <KeyboardAwareScrollView
          style={{marginBottom: isFirstScreen ? 0 : 50 * k}}
          onKeyboardWillShow={frames => setTimeout(() => (this.keyboardHeight = frames.endCoordinates.height), frames.duration)}
          onKeyboardWillHide={() => (this.keyboardHeight = 0)}
        >
          <PhotoArea isFirstScreen={isFirstScreen} onCoverPhoto={this.onCoverPhoto} />
          <ComposeCard isFirstScreen={isFirstScreen} edit={edit} titleBlurred={titleBlurred} ref={r => (this.card = r)} />
          {!isFirstScreen && <EditControls />}
        </KeyboardAwareScrollView>
        {!isFirstScreen && <CreateSaveButton isLoading={this.isLoading} save={this.save} isEnabled={saveable} bottomPadding={this.keyboardHeight} />}
      </Screen>
    );
  }
}

const PhotoArea = observer(({onCoverPhoto, isFirstScreen}) => {
  const backgroundColor = {backgroundColor: isFirstScreen ? colors.LIGHT_GREY : colors.LIGHT_BLUE};
  return botStore.bot.image && botStore.bot.image.source ? (
    <View style={{height: width}}>
      <Image style={{width, height: width}} resizeMode='contain' source={botStore.bot.image && botStore.bot.image.source} />
      <TouchableOpacity onPress={onCoverPhoto} style={styles.changePhotoButton}>
        <RText size={11} weight='Medium' color={colors.DARK_PURPLE} style={{letterSpacing: 0.5}}>
          CHANGE PHOTO
        </RText>
      </TouchableOpacity>
    </View>
  ) : (
    <View style={[styles.imageContainer, backgroundColor]}>
      {!isFirstScreen && (
        <TouchableOpacity onPress={onCoverPhoto} style={{alignItems: 'center'}}>
          <Image source={require('../../../images/iconAddcover.png')} />
          <RText size={14} color={colors.WHITE}>
            Add Cover Photo
          </RText>
        </TouchableOpacity>
      )}
    </View>
  );
});

const CreateSaveButton = observer(({isEnabled, isLoading, save, bottomPadding}) => {
  const {bot} = botStore;
  const buttonText = bot.isNew ? (bot.isPublic ? 'Post' : 'Post (Private)') : 'Save Changes';
  const bottom = bottomPadding - 10;
  return (
    <Button
      style={{bottom, right: 0, left: 0, position: 'absolute', borderRadius: 0, padding: 0, margin: 0}}
      buttonStyle={{padding: 0, margin: 0}}
      isLoading={isLoading}
      isDisabled={!isEnabled}
      onPress={save}
    >
      {buttonText}
    </Button>
  );
});

export default BotCompose;

const styles = StyleSheet.create({
  imageContainer: {
    height: width,
    justifyContent: 'center',
  },
  changePhotoButton: {
    position: 'absolute',
    alignItems: 'center',
    justifyContent: 'center',
    bottom: 20 * k,
    right: 20 * k,
    width: 126 * k,
    height: 30 * k,
    backgroundColor: TRANS_WHITE,
    borderRadius: 2 * k,
  },
});
