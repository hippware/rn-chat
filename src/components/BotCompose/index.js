// @flow

import React from 'react';
import {Alert, TouchableOpacity, Image} from 'react-native';
import {observer, inject, Provider} from 'mobx-react/native';
import {observable} from 'mobx';
import {Actions} from 'react-native-router-flux';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
import {k} from '../Global';
import {colors} from '../../constants';
import Screen from '../Screen';
import Button from '../Button';
import * as log from '../../utils/log';
import {Spinner} from '../common';
import EditControls from './EditControls';
import ComposeCard from './ComposeCard';
import PhotoArea from './BotComposePhoto';
import {settings} from '../../globals';

type Props = {
  botId: string,
  edit?: boolean,
  titleBlurred?: boolean,
};

@inject('wocky')
@observer
class BotCompose extends React.Component<Props> {
  botTitle: ?Object;
  @observable keyboardHeight: number = 0;
  @observable isLoading: boolean = false;
  controls: any;
  @observable bot: Bot;

  // TODO: undo on BotCreate
  // static leftButton = ({edit}) => {
  //   return (
  //     <TouchableOpacity
  //       onPress={() => {
  //         if (edit) {
  //           Alert.alert('Unsaved Changes', 'Are you sure you want to discard the changes?', [
  //             {text: 'Cancel', style: 'cancel'},
  //             {
  //               text: 'Discard',
  //               style: 'destructive',
  //               onPress: () => {
  //                 botStore.bot.load(oldBot);
  //                 Actions.pop();
  //               },
  //             },
  //           ]);
  //         } else {
  //           Actions.pop();
  //         }
  //       }}
  //       style={{marginLeft: 10 * k}}
  //     >
  //       <Image source={require('../../../images/iconBackGrayNew.png')} style={{tintColor: settings.isStaging ? 'rgb(28,247,39)' : 'rgb(117,117,117)'}} />
  //     </TouchableOpacity>
  //   );
  // };

  componentWillMount() {
    if (this.props.botId) {
      this.bot = this.props.wocky.getBot({id: this.props.botId});
    }
  }

  save = async () => {
    const {bot} = this;
    if (!bot.title) {
      // TODO: slide-down notification
      Alert.alert('Title cannot be empty');
      this.botTitle && this.botTitle.focus();
      return;
    }
    try {
      this.isLoading = true;
      const {isNew} = bot;
      await bot.save();
      if (isNew) {
        Actions.pop({animated: false});
        Actions.pop();
        setTimeout(() => Actions.botDetails({item: bot.id, isNew: true}));
      } else {
        Actions.pop();
      }
    } catch (e) {
      // TODO: notificationStore.flash('Something went wrong, please try again.');
      alert('Something went wrong, please try again.');
      log.log('BotCompose save problem', e);
    } finally {
      this.isLoading = false;
    }
  };

  setKeyboardHeight = (frames) => {
    // TODO: more elegant solution...maybe animate the save button position instead of a hard switch?
    setTimeout(() => (this.keyboardHeight = frames.endCoordinates.height), frames.duration);
  };

  scrollToNote = () => {
    if (this.bot.description === '') this.controls.focus();
  };

  render() {
    const {edit, titleBlurred} = this.props;
    const {bot} = this;
    if (!bot) {
      log.log('NO BOT IS DEFINED', {level: log.levels.ERROR});
      return <Screen />;
    }
    const isEnabled = bot.title && bot.title.length > 0 && bot.location && bot.address;

    return (
      <Provider bot={bot}>
        <Screen>
          <KeyboardAwareScrollView style={{marginBottom: 50 * k}} onKeyboardWillShow={this.setKeyboardHeight} onKeyboardWillHide={() => (this.keyboardHeight = 0)}>
            <PhotoArea afterPhotoPost={this.scrollToNote} />
            <ComposeCard edit={edit} titleBlurred={titleBlurred} />
            <EditControls ref={r => (this.controls = r)} />
          </KeyboardAwareScrollView>
          <CreateSaveButton bot={bot} isLoading={this.isLoading} isEnabled={isEnabled} onSave={this.save} bottomPadding={bot.title !== '' ? this.keyboardHeight : 0} />
        </Screen>
      </Provider>
    );
  }
}

const CreateSaveButton = observer(({bot, isEnabled, isLoading, onSave, bottomPadding}) => {
  const buttonText = bot.isNew ? (bot.isPublic ? 'Post' : 'Post (Private)') : 'Save Changes';
  return (
    <Button
      style={{bottom: bottomPadding - 10, right: 0, left: 0, position: 'absolute', borderRadius: 0, padding: 0, margin: 0}}
      buttonStyle={{padding: 0, margin: 0}}
      isDisabled={!isEnabled}
      onPress={onSave}
    >
      {isLoading ? <Spinner color='white' size={22} /> : buttonText}
    </Button>
  );
});

export default BotCompose;
