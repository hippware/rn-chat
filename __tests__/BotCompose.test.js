// @flow

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BotCompose from '../src/components/BotCompose';
import {Provider} from 'mobx-react/native';
// import {} from 'wocky-client';
import injects from './utils/inject-mocks';
import wocky from './utils/mockWocky';

describe('BotCompose', () => {
  it('renders with no data', async () => {
    await wocky.login('user', 'password', 'host');
    const toRender = (
      <Provider wocky={wocky} {...injects}>
        <BotCompose />
      </Provider>
    );
    const tree = renderer.create(toRender).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
