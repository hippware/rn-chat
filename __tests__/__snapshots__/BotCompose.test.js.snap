// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotCompose renders with no data 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <BVLinearGradient
    colors={
      Array [
        "rgb(255,255,255)",
        "rgb(241,242,244)",
        "rgb(243,244,246)",
      ]
    }
    locations={
      Array [
        0,
        0.2,
        1,
      ]
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  />
</View>
`;

exports[`BotCompose renders with owned bot 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <BVLinearGradient
    colors={
      Array [
        "rgb(255,255,255)",
        "rgb(241,242,244)",
        "rgb(243,244,246)",
      ]
    }
    locations={
      Array [
        0,
        0.2,
        1,
      ]
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={undefined}
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutoAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      onKeyboardWillHide={[Function]}
      onKeyboardWillShow={[Function]}
      onScroll={[Function]}
      scrollEventThrottle={0}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "marginBottom": 100,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "white",
              "height": 750,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "bottom": -420,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <MapView
              autoZoom={true}
              fullMap={false}
              initialRegion={
                Object {
                  "latitude": 5,
                  "latitudeDelta": 0.2,
                  "longitude": 5,
                  "longitudeDelta": 0.2,
                }
              }
              location={
                Object {
                  "accuracy": 5,
                  "latitude": 5,
                  "longitude": 5,
                }
              }
              locationStore={
                Object {
                  "start": [Function],
                  "stop": [Function],
                }
              }
              onMarkerSelect={[Function]}
              onPress={[Function]}
              onRegionChangeComplete={[Function]}
              scale={0.5}
              showOnlyBot={true}
              showUser={false}
              style={
                Object {
                  "flex": 1,
                }
              }
              wocky={
                Object {
                  "bots": Object {
                    "storage": Object {
                      "4321": Object {
                        "address": "123 address",
                        "addressData": Object {
                          "city": "",
                          "country": "",
                          "county": "",
                          "state": "",
                        },
                        "description": null,
                        "error": "",
                        "followersSize": 0,
                        "id": "4321",
                        "image": null,
                        "isSubscribed": false,
                        "location": Object {
                          "accuracy": 5,
                          "latitude": 5,
                          "longitude": 5,
                        },
                        "owner": "user",
                        "posts": Object {
                          "count": null,
                          "result": Array [],
                        },
                        "radius": 30,
                        "server": null,
                        "subscribers": Object {
                          "count": null,
                          "result": Array [],
                        },
                        "title": "title",
                        "totalItems": 0,
                        "visibility": 100,
                      },
                    },
                  },
                  "chats": Object {
                    "_list": Array [],
                  },
                  "events": Object {
                    "count": null,
                    "result": Array [],
                  },
                  "files": Object {
                    "storage": Object {},
                  },
                  "geoBots": Object {},
                  "host": "host",
                  "id": "wocky",
                  "password": "password",
                  "profile": Object {
                    "avatar": null,
                    "botsSize": 0,
                    "email": "",
                    "firstName": "",
                    "followedSize": 0,
                    "followersSize": 0,
                    "handle": "",
                    "id": "user",
                    "isBlocked": false,
                    "isFollowed": false,
                    "isFollower": false,
                    "isNew": false,
                    "lastName": "",
                    "loaded": true,
                    "phoneNumber": "",
                    "roles": Array [],
                    "status": "available",
                  },
                  "profiles": Object {
                    "storage": Object {
                      "user": Object {
                        "avatar": null,
                        "botsSize": 0,
                        "firstName": "",
                        "followedSize": 0,
                        "followersSize": 0,
                        "handle": "",
                        "id": "user",
                        "isBlocked": false,
                        "isFollowed": false,
                        "isFollower": false,
                        "isNew": false,
                        "lastName": "",
                        "loaded": false,
                        "roles": Array [],
                        "status": "unavailable",
                      },
                    },
                  },
                  "roster": Object {},
                  "sessionCount": 2,
                  "updates": Array [],
                  "username": "user",
                  "version": "",
                }
              }
            />
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#007bff",
                  "borderBottomWidth": 1,
                  "borderColor": "#006acd",
                  "bottom": null,
                  "left": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateY": -1334,
                    },
                  ],
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "flex": 1,
                      "flexDirection": "row",
                      "padding": 10,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignSelf": "stretch",
                        "flex": 1,
                        "flexDirection": "column",
                        "justifyContent": "center",
                        "marginLeft": 10,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 750,
                "position": "absolute",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "backgroundColor": "rgb(253,95,108)",
                        "borderColor": "rgb(253,95,108)",
                        "borderRadius": 7.2,
                        "borderWidth": 1.2,
                        "height": 203,
                        "overflow": "hidden",
                        "width": 175,
                      }
                    }
                  >
                    <Image
                      collapsable={undefined}
                      resizeMode="contain"
                      source={
                        Object {
                          "testUri": "../../../images/addBotPhoto.png",
                        }
                      }
                      style={
                        Object {
                          "height": 175,
                          "width": 175,
                        }
                      }
                    />
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="middle"
                        numberOfLines={1}
                        style={
                          Array [
                            Object {
                              "color": "rgb(255,255,255)",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 13,
                            },
                            Object {
                              "padding": 2,
                            },
                          ]
                        }
                      >
                        123 address
                      </Text>
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderStyle": "solid",
                          "height": 0,
                          "width": 0,
                        },
                        Object {
                          "borderBottomColor": "transparent",
                          "borderBottomWidth": 0,
                          "borderLeftColor": "transparent",
                          "borderLeftWidth": 5.5,
                          "borderRightColor": "transparent",
                          "borderRightWidth": 5.5,
                          "borderTopColor": "rgb(253,95,108)",
                          "borderTopWidth": 11,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgb(255,255,255)",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 30,
                },
                Array [
                  Object {
                    "padding": 20,
                  },
                  Object {
                    "borderTopColor": "rgba(155,155,155,0.15)",
                    "borderTopWidth": 1,
                  },
                ],
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "paddingLeft": 10,
                    "paddingRight": 30,
                    "width": 30,
                  },
                  Object {
                    "paddingLeft": 28,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../images/faceless.png",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                    "paddingRight": 20,
                  }
                }
              >
                <TextInput
                  autoCorrect={false}
                  blurOnSubmit={false}
                  clearButtonMode="while-editing"
                  maxLength={60}
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Name your bot"
                  placeholderTextColor="rgb(212,212,212)"
                  returnKeyType="done"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                        "height": 50,
                      },
                      Object {
                        "color": "rgb(63,50,77)",
                      },
                    ]
                  }
                  value="title"
                />
              </View>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderTopColor": "rgba(155,155,155,0.15)",
                "borderTopWidth": 1,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "padding": 30,
                  },
                  Object {
                    "borderTopColor": "rgba(155,155,155,0.15)",
                    "borderTopWidth": 1,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "paddingLeft": 10,
                      "paddingRight": 30,
                      "width": 30,
                    },
                    Object {
                      "paddingLeft": 16,
                    },
                  ]
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../images/iconBotLocationPink.png",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                      },
                      Array [
                        Object {
                          "color": "rgb(63,50,77)",
                          "flex": 1,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  123 address
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(255,255,255)",
                },
                Object {
                  "borderTopColor": "rgba(155,155,155,0.15)",
                  "borderTopWidth": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingVertical": 10,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "opacity": 1,
                    "width": 113,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgb(63,50,77)",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                      },
                      Object {
                        "opacity": 0.3,
                      },
                    ]
                  }
                >
                  Private
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 2,
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "rgb(212,212,212)",
                      "borderRadius": 19,
                      "height": 38,
                      "width": 150,
                    }
                  }
                />
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "height": 40,
                      "left": 4,
                      "position": "absolute",
                      "top": 0,
                      "width": 152,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    collapsable={undefined}
                    onMoveShouldSetResponder={[Function]}
                    onMoveShouldSetResponderCapture={[Function]}
                    onResponderEnd={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderReject={[Function]}
                    onResponderRelease={[Function]}
                    onResponderStart={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    onStartShouldSetResponderCapture={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "borderRadius": 15,
                        "flexDirection": "row",
                        "height": 32,
                        "justifyContent": "center",
                        "left": 1,
                        "position": "absolute",
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0.5,
                        "shadowRadius": 0,
                        "top": 5,
                        "transform": Array [
                          Object {
                            "translateX": 69,
                          },
                        ],
                        "width": 75,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../images/iconPublic.png",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "opacity": 1,
                    "width": 113,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgb(63,50,77)",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                      },
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  Public
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "padding": 30,
                  },
                  Object {
                    "borderTopColor": "rgba(155,155,155,0.15)",
                    "borderTopWidth": 1,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "paddingLeft": 10,
                      "paddingRight": 30,
                      "width": 30,
                    },
                    Object {
                      "alignSelf": "flex-start",
                      "paddingLeft": 20,
                      "paddingTop": 14,
                    },
                  ]
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../images/botNotePink.png",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <TextInput
                  maxLength={1500}
                  multiline={true}
                  onChangeText={[Function]}
                  placeholder="Tell us about this place!"
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 15,
                      "height": 400,
                    }
                  }
                  value={null}
                />
              </View>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "borderRadius": 0,
                "borderWidth": 0,
                "bottom": 0,
                "flexDirection": "row",
                "height": 100,
                "justifyContent": "center",
                "left": 0,
                "marginBottom": 10,
                "opacity": 1,
                "position": "relative",
                "right": 0,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                  Array [
                    Object {
                      "color": "white",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 30,
                      "letterSpacing": 0.8,
                    },
                    Object {
                      "color": "rgb(253,95,108)",
                    },
                  ],
                ]
              }
            >
              Cancel Bot
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgb(253,95,108)",
          "borderRadius": 0,
          "borderWidth": 0,
          "bottom": -10,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "left": 0,
          "margin": 0,
          "marginBottom": 10,
          "opacity": 1,
          "padding": 0,
          "position": "absolute",
          "right": 0,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
              "fontSize": 18,
              "textAlign": "center",
            },
            Array [
              Object {
                "color": "white",
                "fontFamily": "Roboto-Regular",
                "fontSize": 30,
                "letterSpacing": 0.8,
              },
              undefined,
            ],
          ]
        }
      >
        Post
      </Text>
    </View>
  </View>
</View>
`;
